chat.php

```diff
@@ -49,15 +49,53 @@ function chat() {
 			$title = $player->rank_name;
 			$staff_level = $player->staff_level;
 			$supported_colors = array(
-				'blue' => -1,
+				'black' => 0,'blue' => -1,
 				'pink' => -2,
 				'gold' => -3,
 				'green' => -4,
 				'teal' => -5,
 				'purple' => -6,
 				'red' => -7,
 			);
-			$user_color = (!$player->forbidden_seal) ? 0 : ($supported_colors[$player->forbidden_seal['color']]);
+
+            // Restrict color if user is no longer able to use it
+            switch($player->chat_color) {
+                case 'blue':
+                case 'pink':
+                    if(!$player->forbidden_seal) {
+                        $player->chat_color = 'black';
+                    }
+                    break;
+                case 'gold':
+                    if(!$player->premium_credits_purchased) {
+                        $player->chat_color = 'black';
+                    }
+                    break;
+                case 'green':
+                    if(!$player->isModerator()) {
+                        $player->chat_color = 'black';
+                    }
+                    break;
+                case 'teal':
+                    if(!$player->isHeadAdmin()) {
+                        $player->chat_color = 'black';
+                    }
+                    break;
+                case 'purple':
+                    if(!$player->isContentAdmin() || !$player->isUserAdmin() || !$player->isHeadAdmin()) {
+                        $player->chat_color = 'black';
+                    }
+                    break;
+                case 'red':
+                    if(!$player->isUserAdmin() || !$player->isHeadAdmin()) {
+                        $player->chat_color = 'black';
+                    }
+                    break;
+                default:
+                    break;
+            }
```

Comment:
Hmm what if we centralize the name colors as defined in premium.php to `User` and then just do an `if(!in_array($player->chat_color, $player->chat_color_options))` check?

https://github.com/elementum-games/shinobi-chronicles/blob/9874185125d13e20f86730561728812046576a11/premium.php#L974-L1029